applications/nrf_desktop/doc/hid_reportq.rst

```diff
@@ -0,0 +1,81 @@
+
+* Maximum number of enqueued HID reports (:ref:`CONFIG_DESKTOP_HID_REPORTQ_MAX_ENQUEUED_REPORTS <config_desktop_app_options>`)
+* Number of supported HID report queues (:ref:`CONFIG_DESKTOP_HID_REPORTQ_QUEUE_COUNT <config_desktop_app_options>`)
+
+See Kconfig help for more details.
+
+Using HID report queue
+**********************
+
+The utility acts as a middleware between a connected HID peripheral and a HID subscriber.
+The following sections describe how to integrate HID report queue APIs in an application module.
+
+The :ref:`nrf_desktop_hid_forward` is an example of an application module that uses the HID report queue utility.
+Refer to the module's implementation (:file:`src/modules/hid_forward.c`) for an example of HID report queue integration.
+
+Allocation
+==========
+
+A HID report queue instance must be allocated with the :c:func:`hid_reportq_alloc` function before it is used.
+A separate HID report queue object needs to be allocated for each HID report subscriber.
+During allocation, the caller specifies the HID subscriber identifier.
+The :c:func:`hid_reportq_get_sub_id` function can be used to access the identifier later.
+After the HID report queue object is no longer used, it should be freed using the :c:func:`hid_reportq_free` function.
+
+HID subscriptions
+=================
+
+A HID report queue instance tracks HID input report subscriptions enabled by a HID subscriber.
+You can use the :c:func:`hid_reportq_subscribe` and :c:func:`hid_reportq_unsubscribe` functions to update the state of enabled subscriptions.
+The functions should be called while handling an :c:struct:`hid_report_subscription_event`.
+You can use the :c:func:`hid_reportq_is_subscribed` function to check if the subscription is enabled for the specified HID input report.
+
+Enqueuing HID input reports
+===========================
+
+If an application module uses the HID report queue instance to locally enqueue HID input reports for a given HID subscriber, every HID report intended for the subscriber should go through the HID report queue.
+An application module can call the :c:func:`hid_reportq_report_add` function for a received HID input report to enqueue the report.
+The function allocates a :c:struct:`hid_report_event` for the received HID input report.
+If a HID subscriber can handle the :c:struct:`hid_report_event`, the event is instantly passed to the subscriber.
+Otherwise, the event is enqueued and will be submitted later.
+
+When a HID subscriber (for example, a USB HID class instance) delivers a HID input report to the HID host (on :c:struct:`hid_report_sent_event`), the :c:struct:`hid_reportq_report_sent` API needs to be called to notify the HID report queue.
+This allows the queue to track the state of HID reports provided to the HID subscriber.
+If there is an enqueued report, the HID report queue object will instantly submit the subsequent :c:struct:`hid_report_event`.
+The enqueued report to be sent is chosen in the round-robin fashion.
```

Comment:
`The enqueued report to be sent is chosen from all report queues in the round-robin fashion.`

Reply:
The subsequent report is chosen from the given HID report queue object only. I would not refer to "internal queues" used per report ID as it might be confusing

Reply:
Then please clarify the pool from which you choose reports to be sent.